internal/providers/github/properties.go

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	v1 "github.com/stacklok/minder/pkg/providers/v1"
+
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+type propertyWrapper func(ctx context.Context, ghCli v1.GitHub, name string) (map[string]any, error)
+
+type propertyOrigin struct {
+	keys    []string
+	wrapper propertyWrapper
+}
+
+type propertyFetcher struct {
+	ghCli v1.GitHub
+
+	propertyOrigins []propertyOrigin
+}
+
+var repoPropertyDefinitions = []propertyOrigin{
+	{
+		keys:    []string{"is_private", "is_archived"},
+		wrapper: getRepoWrapper,
+	},
+	{
+		keys:    []string{"properties.languages"},
+		wrapper: getLanguagesWrapper,
```

Comment:
Not thrilled about this struct, but we can refactor later as long as the interface works for its consumers.